src/index.js

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2000-2017 TeamDev Ltd. All rights reserved.
+ * TeamDev PROPRIETARY and CONFIDENTIAL.
+ * Use is subject to license terms.
+ */
+
+const {
+  Analyzer, FSUrlLoader, PackageUrlResolver, generateAnalysis
+} = require('polymer-analyzer');
+const escape = require('html-escape');
+const fs = require('fs');
+const path = require('path');
+
+module.exports = {
+  generate: argv => {
+    const templateFilePath = path.join(__dirname, "template", "element-doc-template.html");
+    const itemsRootPath = path.resolve(argv[2] || './');
+
+    const analyzer = new Analyzer({
+      urlLoader   : new FSUrlLoader(itemsRootPath),
+      urlResolver : new PackageUrlResolver()
+    });
+
+    analyzer.analyzePackage().then(_package => {
+      const metadata = generateAnalysis(_package, '');
+      let elements;
+
+      const processingPackage = !Boolean(
+          metadata.elements.find(item => item.path.indexOf(path.sep) === -1));
+
+      elements = splitByElements(metadata.elements, itemsRootPath);
+
+      console.log(`Processing ${processingPackage ? "package" : "element"} ` +
+                  `from the ${itemsRootPath} directory`);
+
+      createDocFiles(templateFilePath, elements, itemsRootPath, processingPackage);
+    });
+  }
+};
+
+function splitByElements(rawElementsData, itemsRootPath) {
+  const elements = {};
+
+  rawElementsData.forEach(item => {
+    const elementRootPath =
+        item.path.substring(0, item.path.indexOf(path.sep)) || getElementName(itemsRootPath);
+
+    if (elements[elementRootPath]) {
+      elements[elementRootPath].push(item);
+    } else {
+      elements[elementRootPath] = [item];
+    }
+  });
+
+  return elements;
+}
+
+function createDocFiles(templateFilePath, elements, itemsRootPath, processingPackage) {
+  fs.readFile(templateFilePath, 'utf-8', (err, data) => {
+    if (err) {
+      throw new Error(`Failed to read the template\n${err.message}`);
+    }
+
+    Object.keys(elements).forEach((rootPath, index) => {
+      const docFilePath = processingPackage
+          ? `${itemsRootPath}/${rootPath}/index.html`
+          : `${itemsRootPath}/index.html`;
+
+      const docFileContent = fillContent(data, elements[rootPath], rootPath);
+      fs.writeFile(docFilePath, docFileContent, err => {
+        if (err) {
+          throw new Error(`Failed to fill the doc file\n${err.message}`);
+        }
+        if (index === Object.keys(elements).length - 1) {
+          console.log('Documentation is generated!')
```

Comment:
Hmm, "is generated" looks strange. Something like "has been" or "was" might be more correct, but still sounds odd. What about something "Documentation generation completed" or "Documentation generation successful"? 

BTW, if just one file is generated, it might actually be useful to inform the user about the name of the file that was generated.